src/graphql_client/core.clj

```diff
@@ -2,30 +2,38 @@
 
 (declare args->str)
 
-(defn arg->str [v]
-  (cond (map? v) (str "{" (args->str v) "}")
-        (keyword? v) (str "\"" (name v) "\"")
-        (string? v) (str "\"" v "\"")
-        (coll? v) (str "[" (apply str (interpose "," (map arg->str v))) "]")
-        :else (str v)))
+(defn arg->str
+  ([v] (arg->str name v))
+  ([kw-xform v]
+   (cond (map? v) (str "{" (args->str kw-xform v) "}")
+         (keyword? v) (str "\"" (kw-xform v) "\"")
+         (string? v) (str "\"" v "\"")
+         (coll? v) (str "[" (apply str (interpose "," (map (partial arg->str kw-xform) v))) "]")
+         :else (str v))))
 
-(defn args->str [m]
-  (->> (for [[k v] m]
-         [(name k) ":" (arg->str v)])
-       (interpose ",")
-       (flatten)
-       (apply str)))
+(defn args->str
+  ([m] (args->str name m))
+  ([kw-xform m]
+   (->> (for [[k v] m]
+          [(kw-xform k) ":" (arg->str kw-xform v)])
+        (interpose ",")
+        (flatten)
+        (apply str))))
 
-(defn query->str [q]
-  (if (keyword? q)
-    (name q)
-    (let [f (first q)
-          s (second q)
-          r (if (map? s) (drop 2 q) (rest q))]
-      (println f s r)
-      (apply str (flatten (cond (keyword? f) [(name f)
-                                              (if (map? s) ["(" (args->str s) ")"] "")
-                                              (when (seq r) ["{" (interpose " " (map query->str r)) "}"])]
-                                (vector? f) ["{" (map query->str q) "}"]
-                                :else "")))
-      )))
+(defn query->str
+  ([q] (query->str name q))
+  ([kw-xform q]
```

Comment:
What do you think of giving this option as `:key-fn` like the Clojure json libs do?

```clj
(query->str q :key-fn ->camelCase)
```